ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -61,10 +64,12 @@ public class DefaultLocalPersistenceService implements LocalPersistenceService {
   public DefaultLocalPersistenceService(final DefaultPersistenceConfiguration persistenceConfiguration) {
     if(persistenceConfiguration != null) {
       rootDirectory = persistenceConfiguration.getRootDirectory();
+      isRootDirectoryExists = rootDirectory.exists();
```

Comment:
You really can't be reading state in the constructor. You don't have exclusive access to the directory, and arbitrary amounts of time can elapse (and arbitrary state changes can occur) between this constructor call and the service being started. Consider for example if I create two cache managers, pointing at the same directory, but then initialize the first one, mutate the disk, and then shut it down uncleanly, and then start the second one?